apps/glusterfs/brick_entry.go

```diff
@@ -239,3 +229,46 @@ func (b *BrickEntry) DestroyCheck(db *bolt.DB, executor executors.Executor) erro
 func (b *BrickEntry) TotalSize() uint64 {
 	return b.TpSize + b.PoolMetadataSize
 }
+
+func BrickEntryUpgrade(tx *bolt.Tx) error {
+	err := addVolumeIdInBrickEntry(tx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func addVolumeIdInBrickEntry(tx *bolt.Tx) error {
+	clusters, err := ClusterList(tx)
+	if err != nil {
+		return err
+	}
+	for _, cluster := range clusters {
+		clusterEntry, err := NewClusterEntryFromId(tx, cluster)
+		if err != nil {
+			return err
+		}
+		for _, volume := range clusterEntry.Info.Volumes {
+			volumeEntry, err := NewVolumeEntryFromId(tx, volume)
+			if err != nil {
+				return err
+			}
+			for _, brick := range volumeEntry.Bricks {
+				brickEntry, err := NewBrickEntryFromId(tx, brick)
+				if err != nil {
+					return err
+				}
+				if brickEntry.Info.VolumeId == "" {
+					brickEntry.Info.VolumeId = volume
+					err = brickEntry.Save(tx)
```

Comment:
Nice.  In a way think of the db context as something that can be accomplished immediately, without going to the executor.